Prefer using the MPI module in module scope to avoid visiblity issues.

Also fix a couple Fortran language violations uncovered by this change.

diff -urN MUMPS_5.9.0/src/cana_aux_par.F.orig MUMPS_5.9.0/src/cana_aux_par.F
--- MUMPS_5.9.0/src/cana_aux_par.F.orig	2026-04-28 15:06:25.000000000 +0200
+++ MUMPS_5.9.0/src/cana_aux_par.F	2026-08-28 21:55:40.078402100 +0200
@@ -16,7 +16,11 @@
       USE MUMPS_MEMORY_MOD
       USE MUMPS_ANA_ORD_WRAPPERS
       USE MUMPS_ANA_BLK_M, ONLY : LMATRIX_T, COMPACT_GRAPH_T
+#if defined(MUMPS_PREFER_MPI_MODULE)
+      USE MPI
+#else
       INCLUDE 'mpif.h'
+#endif
       PUBLIC CMUMPS_ANA_F_PAR
       INTERFACE CMUMPS_ANA_F_PAR
       MODULE PROCEDURE CMUMPS_ANA_F_PAR
@@ -656,7 +660,7 @@
       END IF
       CALL MPI_ALLGATHERV ( ORDER(1), VERTLOCNBR, MPI_INTEGER,
      &     ord%PERMTAB(1),
-     &     RCVCNTS(1), FIRST(1), MPI_INTEGER, ord%COMM, IERR )
+     &     RCVCNTS, FIRST, MPI_INTEGER, ord%COMM, IERR )
       DO I=1, N
          ord%PERITAB(ord%PERMTAB(I)) = I
       END DO
@@ -822,7 +826,7 @@
         END IF
         CALL MPI_GATHERV ( ORDER(1), VERTLOCNBR, MPI_INTEGER,
      &     ord%PERMTAB(1),
-     &     RCVCNTS(1), FIRST(1), MPI_INTEGER, MASTER,
+     &     RCVCNTS, FIRST, MPI_INTEGER, MASTER,
      &     ord%COMM_PARAORD, IERR )
       END IF 
       IF (ord%MYID_PARAORD.EQ.MASTER) THEN
@@ -2333,7 +2337,7 @@
      &     MPI_INTEGER8, ord%COMM, IERR)
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       TLEN = 0_8
@@ -2552,7 +2556,7 @@
       IF(MEMCNT .GT. MAXMEM) MAXMEM=MEMCNT
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       I = ceiling(real(NROWS_LOC+1)*1.20E0)
@@ -2857,7 +2861,7 @@
       IF(MEMCNT .GT. MAXMEM) MAXMEM=MEMCNT
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       I = ceiling(real(NROWS_LOC+1)*1.20E0)
diff -urN MUMPS_5.9.0/src/dana_aux_par.F.orig MUMPS_5.9.0/src/dana_aux_par.F
--- MUMPS_5.9.0/src/dana_aux_par.F.orig	2026-04-28 15:06:24.000000000 +0200
+++ MUMPS_5.9.0/src/dana_aux_par.F	2026-08-28 21:51:50.873325000 +0200
@@ -16,7 +16,11 @@
       USE MUMPS_MEMORY_MOD
       USE MUMPS_ANA_ORD_WRAPPERS
       USE MUMPS_ANA_BLK_M, ONLY : LMATRIX_T, COMPACT_GRAPH_T
+#if defined(MUMPS_PREFER_MPI_MODULE)
+      USE MPI
+#else
       INCLUDE 'mpif.h'
+#endif
       PUBLIC DMUMPS_ANA_F_PAR
       INTERFACE DMUMPS_ANA_F_PAR
       MODULE PROCEDURE DMUMPS_ANA_F_PAR
@@ -656,7 +660,7 @@
       END IF
       CALL MPI_ALLGATHERV ( ORDER(1), VERTLOCNBR, MPI_INTEGER,
      &     ord%PERMTAB(1),
-     &     RCVCNTS(1), FIRST(1), MPI_INTEGER, ord%COMM, IERR )
+     &     RCVCNTS, FIRST, MPI_INTEGER, ord%COMM, IERR )
       DO I=1, N
          ord%PERITAB(ord%PERMTAB(I)) = I
       END DO
@@ -822,7 +826,7 @@
         END IF
         CALL MPI_GATHERV ( ORDER(1), VERTLOCNBR, MPI_INTEGER,
      &     ord%PERMTAB(1),
-     &     RCVCNTS(1), FIRST(1), MPI_INTEGER, MASTER,
+     &     RCVCNTS, FIRST, MPI_INTEGER, MASTER,
      &     ord%COMM_PARAORD, IERR )
       END IF 
       IF (ord%MYID_PARAORD.EQ.MASTER) THEN
@@ -2333,7 +2337,7 @@
      &     MPI_INTEGER8, ord%COMM, IERR)
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       TLEN = 0_8
@@ -2552,7 +2556,7 @@
       IF(MEMCNT .GT. MAXMEM) MAXMEM=MEMCNT
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       I = ceiling(dble(NROWS_LOC+1)*1.20D0)
@@ -2857,7 +2861,7 @@
       IF(MEMCNT .GT. MAXMEM) MAXMEM=MEMCNT
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       I = ceiling(dble(NROWS_LOC+1)*1.20D0)
diff -urN MUMPS_5.9.0/src/sana_aux_par.F.orig MUMPS_5.9.0/src/sana_aux_par.F
--- MUMPS_5.9.0/src/sana_aux_par.F.orig	2026-08-28 21:07:40.771404900 +0200
+++ MUMPS_5.9.0/src/sana_aux_par.F	2026-08-28 21:29:12.315109000 +0200
@@ -16,7 +16,11 @@
       USE MUMPS_MEMORY_MOD
       USE MUMPS_ANA_ORD_WRAPPERS
       USE MUMPS_ANA_BLK_M, ONLY : LMATRIX_T, COMPACT_GRAPH_T
+#if defined(MUMPS_PREFER_MPI_MODULE)
+      USE MPI
+#else
       INCLUDE 'mpif.h'
+#endif
       PUBLIC SMUMPS_ANA_F_PAR
       INTERFACE SMUMPS_ANA_F_PAR
       MODULE PROCEDURE SMUMPS_ANA_F_PAR
@@ -656,7 +660,7 @@
       END IF
       CALL MPI_ALLGATHERV ( ORDER(1), VERTLOCNBR, MPI_INTEGER,
      &     ord%PERMTAB(1),
-     &     RCVCNTS(1), FIRST(1), MPI_INTEGER, ord%COMM, IERR )
+     &     RCVCNTS, FIRST, MPI_INTEGER, ord%COMM, IERR )
       DO I=1, N
          ord%PERITAB(ord%PERMTAB(I)) = I
       END DO
@@ -822,7 +826,7 @@
         END IF
         CALL MPI_GATHERV ( ORDER(1), VERTLOCNBR, MPI_INTEGER,
      &     ord%PERMTAB(1),
-     &     RCVCNTS(1), FIRST(1), MPI_INTEGER, MASTER,
+     &     RCVCNTS, FIRST, MPI_INTEGER, MASTER,
      &     ord%COMM_PARAORD, IERR )
       END IF 
       IF (ord%MYID_PARAORD.EQ.MASTER) THEN
@@ -2333,7 +2337,7 @@
      &     MPI_INTEGER8, ord%COMM, IERR)
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       TLEN = 0_8
@@ -2552,7 +2556,7 @@
       IF(MEMCNT .GT. MAXMEM) MAXMEM=MEMCNT
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       I = ceiling(real(NROWS_LOC+1)*1.20E0)
@@ -2857,7 +2861,7 @@
       IF(MEMCNT .GT. MAXMEM) MAXMEM=MEMCNT
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       I = ceiling(real(NROWS_LOC+1)*1.20E0)
diff -urN MUMPS_5.9.0/src/zana_aux_par.F.orig MUMPS_5.9.0/src/zana_aux_par.F
--- MUMPS_5.9.0/src/zana_aux_par.F.orig	2026-04-28 15:06:26.000000000 +0200
+++ MUMPS_5.9.0/src/zana_aux_par.F	2026-08-28 21:59:38.658456600 +0200
@@ -16,7 +16,11 @@
       USE MUMPS_MEMORY_MOD
       USE MUMPS_ANA_ORD_WRAPPERS
       USE MUMPS_ANA_BLK_M, ONLY : LMATRIX_T, COMPACT_GRAPH_T
+#if defined(MUMPS_PREFER_MPI_MODULE)
+      USE MPI
+#else
       INCLUDE 'mpif.h'
+#endif
       PUBLIC ZMUMPS_ANA_F_PAR
       INTERFACE ZMUMPS_ANA_F_PAR
       MODULE PROCEDURE ZMUMPS_ANA_F_PAR
@@ -656,7 +660,7 @@
       END IF
       CALL MPI_ALLGATHERV ( ORDER(1), VERTLOCNBR, MPI_INTEGER,
      &     ord%PERMTAB(1),
-     &     RCVCNTS(1), FIRST(1), MPI_INTEGER, ord%COMM, IERR )
+     &     RCVCNTS, FIRST, MPI_INTEGER, ord%COMM, IERR )
       DO I=1, N
          ord%PERITAB(ord%PERMTAB(I)) = I
       END DO
@@ -822,7 +826,7 @@
         END IF
         CALL MPI_GATHERV ( ORDER(1), VERTLOCNBR, MPI_INTEGER,
      &     ord%PERMTAB(1),
-     &     RCVCNTS(1), FIRST(1), MPI_INTEGER, MASTER,
+     &     RCVCNTS, FIRST, MPI_INTEGER, MASTER,
      &     ord%COMM_PARAORD, IERR )
       END IF 
       IF (ord%MYID_PARAORD.EQ.MASTER) THEN
@@ -2333,7 +2337,7 @@
      &     MPI_INTEGER8, ord%COMM, IERR)
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       TLEN = 0_8
@@ -2552,7 +2556,7 @@
       IF(MEMCNT .GT. MAXMEM) MAXMEM=MEMCNT
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       I = ceiling(dble(NROWS_LOC+1)*1.20D0)
@@ -2857,7 +2861,7 @@
       IF(MEMCNT .GT. MAXMEM) MAXMEM=MEMCNT
       CALL MUMPS_I8DEALLOC(SNDCNT, MEMCNT=MEMCNT)
       RDISPL(:) = MAXS
-      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL(1), 
+      CALL MPI_REDUCE_SCATTER ( SIPES(1,1), LENG(1), RDISPL, 
      &     MPI_INTEGER, MPI_SUM, ord%COMM, IERR )
       DEALLOCATE(SIPES)
       I = ceiling(dble(NROWS_LOC+1)*1.20D0)
